### DISCUSSION TITLE
Vitest (+ react testing library tests)

### DISCUSSION THREAD

**Discussion by jkone27** [nojaf/vite-plugin-fable#12] [Q&A]
This plugin is dope :) 

how to run tests for vite but in F#? is that possible , e.g. say vitest and react-testing-library do we need to add 
Fable Jester as dependency to do so, 

i believe vitest has mostly compatible with jest declarations? 
maybe would be better to get vitest tyopes translated with glutinum cli?

just trying to figure out what i would need to also add test to a Feliz template using this plugin, thank you so much

**Comment by nojaf** [comment]
I have never used Vitest, so I have no idea.
If it calls all the right plugin hooks, I assume it will probably work.
You will have to investigate yourself to find out.
I would be ok to accept sensible modifications if you need them.

**Comment by laurentpayot** [comment]
@jkone27 

- [Fable.Mocha](https://github.com/Zaid-Ajaj/Fable.Mocha) are Mocha tests for F#
- [Jest tests should be compatible with Mocha](https://jestjs.io/docs/migration-guide#:~:text=Jasmine%20like%20API%20(for%20example%20Mocha)%2C%20Jest%20should%20be%20mostly%20compatible)
- [Vitest has been designed with a Jest compatible API](https://vitest.dev/guide/migration#migrating-from-jest)

So _maybe_ you should be able to run your Fable.Mocha tests in Vitest. That’s what I am going to try in the next few days, so stay tuned :wink:

**Reply by laurentpayot** [reply]
I confirm Vitest is working with (basic) tests generated with Fable.Mocha :tada:  Just set `globals` to true in your  Vite config to use the APIs globally like Jest:

```js
export default defineConfig({
  test: {
    globals: true,
  },
})
```
